assets/src/web-stories-block/block/block-types/selected-stories/selectStories.js

```diff
@@ -0,0 +1,335 @@
+    }
+  }
+`;
+
+const StoryGrid = styled.div(
+  ({ columnWidth, theme }) => `
+  display: grid;
+  overflow: auto !important;
+  width: 100%;
+  height: calc(100% - 95px) !important;
+  padding: 10px 5px;
+  grid-column-gap: ${theme.DEPRECATED_THEME.grid.columnGap.desktop}px;
+  grid-row-gap: 80px;
+  grid-template-columns: repeat(auto-fill, ${columnWidth - 5}px);
+  grid-template-rows: auto !important;
+  scroll-margin-top: 30vh;
+  margin-top: 2px; // this is for keyboard focus
+
+  ${theme.DEPRECATED_THEME.breakpoint.tablet} {
+    grid-column-gap: ${theme.DEPRECATED_THEME.grid.columnGap.tablet}px;
+  }
+  ${theme.DEPRECATED_THEME.breakpoint.largeDisplayPhone} {
+    grid-column-gap: ${
+      theme.DEPRECATED_THEME.grid.columnGap.largeDisplayPhone
+    }px;
+  }
+  ${theme.DEPRECATED_THEME.breakpoint.smallDisplayPhone} {
+    grid-column-gap: ${
+      theme.DEPRECATED_THEME.grid.columnGap.smallDisplayPhone
+    }px;
+  }
+  ${theme.DEPRECATED_THEME.breakpoint.min} {
+    grid-column-gap: ${theme.DEPRECATED_THEME.grid.columnGap.min}px;
+  }
+`
+);
+
+const SearchContainer = styled.div`
+  display: inline-block;
+  vertical-align: baseline;
+  position: relative;
+  width: 100%;
+  height: 29px;
+  @media ${({ theme }) => theme.DEPRECATED_THEME.breakpoint.smallDisplayPhone} {
+    left: ${({ theme }) =>
+      `${theme.DEPRECATED_THEME.standardViewContentGutter.min}px`};
+    max-width: 100%;
+    justify-content: flex-start;
+  }
+`;
+
+const SearchInner = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  width: min(${DASHBOARD_LEFT_NAV_WIDTH}px, 100%);
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const DropdownContainer = styled.div`
+  margin: auto 8px;
+  align-self: flex-end;
+`;
+
+const AuthorDropdown = styled(Dropdown)`
+  & > div {
+    max-height: 350px;
+    overflow-x: hidden;
+    overflow-y: auto;
+    border-radius: 8px;
+    box-shadow: 0px 4px 14px rgba(0, 0, 0, 0.25);
+
+    & > div {
+      box-shadow: none;
+    }
+  }
+`;
+
+const DetailRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+`;
+
+function SelectStories({
+  selectedStories,
+  orderedStories,
+  pageSize,
+  search,
+  currentAuthor,
+  setCurrentAuthor,
+  sort,
+  addItemToSelectedStories,
+  removeItemFromSelectedStories,
+  allPagesFetched,
+  isLoading,
+  page,
+}) {
+  const [authors, setAuthors] = useState([]);
+  const [debouncedTypeaheadChange] = useDebouncedCallback((value) => {
+    search.setKeyword(value);
+  }, TEXT_INPUT_DEBOUNCE);
+
+  const onSortChange = useCallback(
+    (newSort) => {
+      sort.set(newSort);
+    },
+    [sort]
+  );
+
+  useEffect(() => {
+    const items = [
+      {
+        label: __('All authors', 'web-stories'),
+        value: '0',
+      },
+    ];
+
+    global.webStoriesBlockSettings.authors.forEach((author) => {
+      items.push({
+        label: author.display_name,
+        value: author.ID.toString(),
+      });
+    });
```

Comment:
As noted before, please not do this. Please use a call to the rest api user endpoint using the `-1` param. User tables can get large and doing this how you currently have is problematic.

Reply:
To be fair, doing such a call with `-1` is also not ideal, because it results in huuuuuuge dropdowns. Better to use the author dropdown component from Gutenberg if easily possible. It has a search function if there are too many users.

Reply:
I agree with both of your comments. I'm working on replacing it with a custom component that is based on the `TypeaheadSearch` from the dashboard, but it's a bit flaky right now, I'm fixing it up and hoping to update the PR soon, with other review points as well.

But again the `Selected Stories` block type uses multiple components from the `dashboard`, and as @spacedmonkey pointed out that we should have separate API providers for the blocks, do we want to avoid relying on the dashboard for anything related to block?

Reply:
@divyarajmasani Have you seen this component here: https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/post-author/index.js? I don't want us to reinvent the wheel.

Left a comment on the second question in the other thread.

Reply:
I'll check the post-author component.

Reply:
We need to add author selection to filter stories in the story picker modal, while the `post-author` component updates the author of the current post. I think we can't directly use the same component.

Reply:
Then can we copy that component and modify it to our needs?